examples/src/main/java/tachyon/examples/JournalCrashTest.java

```diff
@@ -0,0 +1,311 @@
+          try {
+            sTfs.open(checkURI);
+          } catch (IOException ioe) {
+            // File not exist. This is unexpected for CREATE_FILE.
+            return false;
+          }
+        } else if (ClientOpType.CREATE_DELETE_FILE == opType) {
+          try {
+            sTfs.open(checkURI);
+          } catch (IOException ioe) {
+            // File not exist. This is expected for CREATE_DELETE_FILE.
+            continue;
+          }
+          return false;
+        } else if (ClientOpType.CREATE_RENAME_FILE == opType) {
+          try {
+            sTfs.open(new TachyonURI(checkURI + "-rename"));
+          } catch (IOException ioe) {
+            // File not exist. This is unexpected for CREATE_RENAME_FILE.
+            return false;
+          }
+        }
+        //else if (ClientOpType.CREATE_TABLE == opType) {
+        //  if (tfs.getRawTable(new TachyonURI(workDir + s)).getId() == -1) {
+        //    tfs.close();
+        //    return false;
+        //  }
+        //}
+      }
+    }
+    return true;
+  }
+
+  public static void main(String[] args) {
+    // Parse the input args.
+    if (!parseInputArgs(args)) {
+      printUsage();
+      System.exit(EXIT_FAILED);
+    }
+
+    // Set NO_STORE and NO_PERSIST so that this test can work without TachyonWorker.
+    sClientOptions = new ClientOptions.Builder(new TachyonConf())
+        .setStorageTypes(TachyonStorageType.NO_STORE, UnderStorageType.NO_PERSIST).build();
+    sClientThreadList = new ArrayList<ClientThread>();
+    sTfs = TachyonFileSystem.get();
+    try {
+      sTfs.delete(sTfs.open(new TachyonURI(sTestDir)));
+    } catch (IOException ioe) {
+      // Test Directory not exist
+    }
+
+    for (int i = 0; i < sCreateFileClientNum; i ++) {
+      ClientThread thread = new ClientThread(sTestDir + "/createFile" + i + "/",
+          ClientOpType.CREATE_FILE);
+      sClientThreadList.add(thread);
+    }
+    for (int i = 0; i < sCreateDeleteClientNum; i ++) {
+      ClientThread thread = new ClientThread(sTestDir + "/createDelete" + i + "/",
+          ClientOpType.CREATE_DELETE_FILE);
+      sClientThreadList.add(thread);
+    }
+    for (int i = 0; i < sCreateRenameClientNum; i ++) {
+      ClientThread thread = new ClientThread(sTestDir + "/createRename" + i + "/",
+          ClientOpType.CREATE_RENAME_FILE);
+      sClientThreadList.add(thread);
+    }
+
+    // Launch all the client threads and wait for them. If Master crashes, all the threads will
+    // stop at a certain time.
+    for (Thread thread : sClientThreadList) {
+      thread.start();
+    }
+    for (Thread thread : sClientThreadList) {
+      try {
+        thread.join();
+      } catch (InterruptedException e) {
+        LOG.error("Error when waiting thread", e);
+      }
+    }
+
+    // Restart Tachyon Master.
+    String restartMasterCommand = new TachyonConf().get(Constants.TACHYON_HOME)
+        + "/bin/tachyon-start.sh master";
```

Comment:
It is a little confusing that the master is started in the bash script, but also started here. Why is the master started in both places?

Ideally, it should only be started/killed in one place.